### PR TITLE
add .tif extension if missing for tif image file

### PR DESCRIPTION
**Pull request by bingzhang**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,9 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
+### Fixed
+- Fix the tiff file extension. 
+
```
Comment (robkooper) [inline CHANGELOG.md:9]: don't forget to bump the version number in the extractor_info.json for geotiff.

File: preview.geotiff/ncsa.geo.tiff.py [hunk 2]
```diff
@@ -70,9 +70,19 @@ def check_message(self, connector, host, secret_key, resource, parameters):
     # Process the file and upload the results
     def process_message(self, connector, host, secret_key, resource, parameters):
         self.logger = logging.getLogger(__name__)
-
+        isTiffType = False
+        if parameters.get('source'):
+            mimetype = parameters.get('source').get('mimeType')
+            if mimetype == 'image/tiff' or mimetype == 'image/tif':
+                isTiffType = True
         """Process the geotiff and create geoserver layer"""
```
Comment (robkooper) [inline preview.geotiff/ncsa.geo.tiff.py]: this is the python doc string, and there should not be code between this and the def

File: preview.geotiff/ncsa.geo.tiff.py [hunk 3]
```diff
@@ -70,9 +70,19 @@ def check_message(self, connector, host, secret_key, resource, parameters):
...
         filename = resource['name']
+        # add .tif to filename if filename doesnot have it.
+        if isTiffType:
+            extension = os.path.splitext(filename)[1]
```
Comment (robkooper) [inline preview.geotiff/ncsa.geo.tiff.py]: make sure to use lowercase in case the file is "FOO.TIF"

**Review by robkooper — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (bingzhang) [reply CHANGELOG.md:9]: fixed

File: preview.geotiff/ncsa.geo.tiff.py (hunk 3 above)
Comment (bingzhang) [reply preview.geotiff/ncsa.geo.tiff.py]: fixed

File: preview.geotiff/ncsa.geo.tiff.py (hunk 2 above)
Comment (bingzhang) [reply preview.geotiff/ncsa.geo.tiff.py]: fixed

**Review by robkooper — APPROVED** [review]

**Pull request merged**